src/runtime_src/core/tools/common/XBUtilities.cpp

```diff
@@ -445,11 +445,11 @@ XBUtilities::get_axlf_section(const std::string& filename, axlf_section_kind kin
   if (!section)
     throw std::runtime_error("Section not found");
 
-  auto buf = new char[section->m_sectionSize];
+  std::unique_ptr<char> buf(new char[section->m_sectionSize]);
   in.seekg(section->m_sectionOffset);
-  in.read(buf, section->m_sectionSize);
+  in.read(buf.get(), section->m_sectionSize);
 
-  return std::make_pair(buf, section->m_sectionSize);
+  return std::make_pair(buf.get(), section->m_sectionSize);

```

Comment:
No that wont work.  The unique_ptr goes out of scope at lone 453 and the returned buffer has been deleted.  You will have to pass ownership to the caller via std::unique_ptr in the returned pair.